dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHelper.java

```diff
@@ -0,0 +1,42 @@
+package io.dockstore.webservice.helpers;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.dockstore.webservice.Constants.LAMBDA_FAILURE;
+
+/**
+ * A helper class for handling git related problems
+ * @author aduncan
+ */
+public final class GitHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GitHelper.class);
+
+    private GitHelper() {
+
+    }
+
+    /**
+     * Parse git references (ex. refs/heads/feature/foobar) and returns a reference name
+     * @param gitReference Git Reference of the form refs/tags/name or refs/heads/name
+     * @return Git reference name
+     */
+    public static String parseGitHubReference(String gitReference) {
```

Comment:
I would have this return an Optional<String>, and make the caller responsible for throwing an exception. Seems odd to have lambda exception being thrown from a method that parses a string.